pkg/controller/backup_sync_controller_test.go

```diff
@@ -154,3 +159,107 @@ func TestBackupSyncControllerRun(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteUnused(t *testing.T) {
+	tests := []struct {
+		name               string
+		getAllBackupsError error
+		cloudBackups       []*v1.Backup
+		k8sBackups         []*arktest.TestBackup
+		namespace          string
+	}{
+		{
+			name: "normal case",
+			cloudBackups: []*v1.Backup{
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").Backup,
+				// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").Backup,
+				// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").Backup,
+			},
+			namespace: "ns-1",
+			k8sBackups: []*arktest.TestBackup{
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupA").WithPhase(v1.BackupPhaseCompleted),
+				// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupB").WithPhase(v1.BackupPhaseCompleted),
+				// arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupC").WithPhase(v1.BackupPhaseCompleted),
+			},
+		},
+		// todo: add more test cases
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			var (
+				bs              = &arktest.BackupService{}
+				client          = fake.NewSimpleClientset()
+				sharedInformers = informers.NewSharedInformerFactory(client, 0)
+				logger          = arktest.NewLogger()
+			)
+
+			c := NewBackupSyncController(
+				client.ArkV1(),
+				bs,
+				"bucket",
+				time.Duration(0),
+				test.namespace,
+				sharedInformers.Ark().V1().Backups(),
+				logger,
+			).(*backupSyncController)
+
+			// bs.On("GetAllBackups", "bucket").Return(test.cloudBackups, test.getAllBackupsError)
+
+			// get client backups
+			cloudBackupNames := sets.NewString()
+			for _, cloudBackup := range test.cloudBackups {
+				cloudBackupNames.Insert(cloudBackup.Name)
+			}
+
+			// insert additional backups into Kubernetes, if any
+			for _, backup := range test.k8sBackups {
+				if test.k8sBackups != nil {
+					err := sharedInformers.Ark().V1().Backups().Informer().GetStore().Add(backup.Backup)
+					if err != nil {
+						fmt.Println("Error adding k8s backup: ", err)
+						return
+					}
+				}
+
+				// assert newly inserted backups exist in Kubernetes
+				assert.True(t, backupsExist(t, c, test.k8sBackups))
```

Comment:
You don't need to do this - you added them to the informer, and that's sufficient.